exporting/graphite/graphite.c

```diff
@@ -51,6 +51,74 @@ int init_graphite_instance(struct instance *instance)
     return 0;
 }
 
+static inline void fix_graphite_label_key(char *dst, char *src, size_t len) {
+    while (*src != '\0' && len) {
+        if (*src == ';' || *src == '!' || *src == '^' || *src == '=')
+            *dst++ = '_';
+        else
+            *dst++ = *src;
+        src++;
+        len--;
+    }
+    *dst = '\0';
+}
+
+static inline void fix_graphite_label_value(char *dst, char *src, size_t len) {
+    while (*src != '\0' && len) {
+        if (*src == ';' || *src == '~')
+            *dst++ = '_';
+        else
+            *dst++ = *src;
+        src++;
+        len--;
+    }
+    *dst = '\0';
+}
+
+/**
+ * Format host labels for JSON connector
+ *
+ * @param instance an instance data structure.
+ * @param host a data collecting host.
+ * @return Always returns 0.
+ */
+int format_host_labels_graphite_plaintext(struct instance *instance, RRDHOST *host)
+{
+    if (!instance->labels)
+        instance->labels = buffer_create(1024);
+
+    if (!instance->labels)
```

Comment:
A suggestion, instead to have this condition here, you could move it for line 89 and put `{ }` for the if, because this condition only makes sense after you try to allocate again, of course, this suggestion can be ignored case this function is called only one time.

Reply:
You are right. If there is a memory allocating error `callocz()` and `mallocz()` call `fatal()` so there is no need to check the result.